code/search/SearchIntrospection.php

```diff
@@ -41,18 +43,19 @@ public static function hierarchy($class, $includeSubclasses = true, $dataOnly =
                 $classes = array_unique(array_merge($classes, array_values(ClassInfo::subclassesFor($class))));
             }
 
-            $idx = array_search('DataObject', $classes);
+            $idx = array_search('SilverStripe\ORM\DataObject', $classes);
             if ($idx !== false) {
                 array_splice($classes, 0, $idx+1);
             }
 
-            if ($dataOnly) {
-                foreach ($classes as $i => $class) {
-                    if (!DataObject::has_own_table($class)) {
-                        unset($classes[$i]);
-                    }
-                }
-            }
+//@todo find another way to determine if a dataobject does not have a table
+//            if ($dataOnly) {
+//                foreach ($classes as $i => $class) {
+//                    if (!DataObject::has_own_table($class)) {
```

Comment:
I originally left this @todo in my initial round of work because I wasn't aware of the new SS4 method of determining a table. We should uncomment the lines and use `DataObject::getSchema()->classHasTable($class)` instead